wallets/validate.js

```diff
@@ -0,0 +1,110 @@
+/*
+   we want to take all the validate members from the provided wallet
+   and compose into a single yup schema for formik validation ...
+   the validate member can be on of:
+   - a yup schema
+   - a function that throws on an invalid value
+   - a regular expression that must match
+*/
+
+import { autowithdrawSchemaMembers, vaultEntrySchema } from '@/lib/validate'
+import * as Yup from '@/lib/yup'
+import { canReceive } from './common'
+
+export default async function validateWallet (walletDef, data,
+  { yupOptions = { abortEarly: true }, topLevel = true, serverSide = false, skipGenerated = false } = {}) {
+  let schema = composeWalletSchema(walletDef, serverSide, skipGenerated)
+
+  if (canReceive({ def: walletDef, config: data })) {
+    schema = schema.concat(autowithdrawSchemaMembers)
+  }
+
+  await schema.validate(data, yupOptions)
+
+  const casted = schema.cast(data, { assert: false })
+  if (topLevel && walletDef.validate) {
```

Comment:
We added an optional top level validation to wallet modules so an entire wallet could be validated in aggregate.